tests/Unit/Domain/CoordinateMaps/Test_Composition.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <random>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/Composition.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.tpp"
+#include "Domain/ElementToBlockLogicalMap.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain::CoordinateMaps {
+
+namespace {
+void test_composition() {
+  INFO("Composition");
+
+  using Affine2D = ProductOf2Maps<Affine, Affine>;
+  Parallel::register_classes_with_charm<
+      CoordinateMap<Frame::ElementLogical, Frame::BlockLogical, Affine2D>,
+      CoordinateMap<Frame::BlockLogical, Frame::Inertial, Affine2D>>();
+
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<double> logical_dist{-1., 1.};
+
+  // Domain is [0, 1] x [1, 3], and first dim is split in half, so element is
+  // [0, 0.5] x [1, 3]
+  const ElementId<2> element_id{0, {{{1, 0}, {0, 0}}}};
+  const Composition<
+      tmpl::list<Frame::ElementLogical, Frame::BlockLogical, Frame::Inertial>,
+      2>
+      map{element_to_block_logical_map(element_id),
+          std::make_unique<
+              CoordinateMap<Frame::BlockLogical, Frame::Inertial, Affine2D>>(
+              Affine2D{{-1., 1., 0., 1.}, {-1., 1., 1., 3.}})};
+
+  {
+    INFO("Semantics");
+    test_serialization(map);
+    test_copy_semantics(map);
+    auto move_map = map;
+    test_move_semantics(std::move(move_map), map);
+  }
+
+  {
+    INFO("Properties");
+    const auto& maps = map.maps();
+    CHECK(*get<0>(maps) == *element_to_block_logical_map(element_id));
+    CHECK((map.get_component<Frame::ElementLogical, Frame::BlockLogical>() ==
+           *get<0>(maps)));
+  }
+
+  CHECK_FALSE(map.is_identity());
+  CHECK_FALSE(map.inv_jacobian_is_time_dependent());
+  CHECK_FALSE(map.jacobian_is_time_dependent());
+
+  const auto xi =
+      make_with_random_values<tnsr::I<DataVector, 2, Frame::ElementLogical>>(
+          make_not_null(&generator), make_not_null(&logical_dist),
+          DataVector(5));
+  const auto x = map(xi);
+  const auto jacobian = map.jacobian(xi);
```

Comment:
given the non-trivial math being done in the implementation, can you test a 3d map with a full jacobian? At least that jac * inv_jac == identity. Seems tough to get that property to hold by luck :)